### ISSUE TITLE
Can't compile?

### ISSUE THREAD

**Issue by thislooksfun** [hisaac/LEDSA#1]
Bit of a silly question, but how do you get it to compile to a usable .bundle file? It just gives me "The run destination My Mac is not valid for Running the scheme 'iTunesX Visualizer (Upgraded)'."

**Comment by hisaac** [comment]
I unfortunately cannot help you with the problem. I'm not a Mac developer, so I don't have experience debugging this type of thing.

I grabbed this code from the original developer's website before his site went down, and have uploaded it here more for archival purposes than anything else. I'll leave the issue open though if someone else figures out what's wrong and can fix it. Let me know though if you figure it out!

I also updated the project's readme to more clearly outline the state of the project.

**Comment by thislooksfun** [comment]
Ah, fair enough. I'll see what I can figure out in my free time. If anyone else finds something first, please don't hesitate to share.

**Comment by albert-rutkowski** [comment]
Go to : project - ledsa > Build Settings [tab] > Base SDK and change it for Latest OS X...
Build...
And now I've got 4 warnings and 1 error

/Users/alberto/Developer/LED-Spectrum-Analyser/iTuneVisualAPI/iTunesVisualAPI.h:131:2: Unknown type name 'SoundComponentData'
SoundComponentData				soundFormat;			/* Input */

**Comment by hisaac** [comment]
I know that the plugin stopped working in the most recent iTunes and/or macOS update (not sure which one broke it), so that could be related to why it's not building in Xcode? ¯\\\_(ツ)\_/¯

**Comment by albert-rutkowski** [comment]
The recent update has broken it. I know because I'm using it.
It's not building in Xcode because the newest Xcode don't support such old OS X releases.
And the version of iTunesVisualAPI has changed too... or not...
I'm trying to repair this.
Root cause is that in Carbon some core functions is deprecated.

**Comment by thislooksfun** [comment]
Yeah, I wanted to update it, but it's kinda hard to do when it doesn't compile at all.

**Comment by donly** [comment]
It does **not** work properly on iTunes 12.6.1.x macOS 10.12.5 (16F73) now.

I have fork this project, after cloning it  I found this is not the latest source code,  it it version 2.0.6, and the latest version is 3.0.6.

> History
> - January 2014. Adds ability to use cover art as background image. Cover art colours feature tweaked a little.

So sad but true.

**Comment by ghost** [comment]
[LED Spectrum Analyser.bundle.zip](https://github.com/hisaac/LED-Spectrum-Analyser/files/1117906/LED.Spectrum.Analyser.bundle.zip)
I only have this (latest version), does it help?

**Comment by hisaac** [comment]
I've emailed Graham—the original developer—to see if we can get the source code for the most recent version. 🤞

**Comment by donly** [comment]
Thanks @iDjay, 
Your version is 3.0.7 and it's newer than 3.0.6.
Unfortunately, It can't work properly on iTunes 12.6.1.x macOS 10.12.5 (16F73) as well.

And thanks to @hisaac, you're so kind a man. 👍

**Comment by gingerbeardman** [comment]
Version 3.0.7 from releases page works for me in iTunes 12.9.5.5 on Mojave.

Required selecting a few times.

By that I mean it was initially a black window but after selecting some other visualizer and switching back to LED a few times it started to work.

> <img width="1080" alt="Screen shot 2020-08-31 at 12 01 44" src="https://user-images.githubusercontent.com/49612/91713510-d2502e00-eb81-11ea-9675-abe04e58d88f.png">

Here are the visualizers I have installed, they all work:

> <img width="244" alt="Screen shot 2020-08-31 at 12 03 23" src="https://user-images.githubusercontent.com/49612/91713598-f9a6fb00-eb81-11ea-98d3-7d971a1e98c0.png">

**Comment by donly** [comment]
It still work on Catalina on Music app version `1.0.6.10`, everything goes well except delay a few seconds.

![37661598920559_ pic_hd](https://user-images.githubusercontent.com/848691/91782071-afc81e80-ec2e-11ea-8b27-cc53ec26db07.jpg)

**Comment by hisaac** [comment]
Well how about that! I wonder why it stopped working and then started again.

**Comment by gingerbeardman** [comment]
> I wonder why it stopped working and then started again.

Apple 😂

**Comment by donly** [comment]
> I've emailed Graham—the original developer—to see if we can get the source code for the most recent version. 🤞

Hi, @hisaac. Any feedback there? I hope the author can update to support Big Sur, I can buy it from somewhere as I like this awesome project very much.

**Comment by hisaac** [comment]
Sadly, I never received a response @donly.
